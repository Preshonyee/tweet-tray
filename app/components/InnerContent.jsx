/* eslint indent: 0 */
import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'styled-components';

import * as constants from '../constants';

export const InnerContentStyle = Styled.section`
  overflow-x: hidden;
  overflow-y: auto;
  padding: ${constants.SPACING}px;
  width: auto;

  &::-webkit-scrollbar-track {
    background-color ${(props) => {
      return props.theme === 'day' ? constants.WHITE : constants.DARK_MODE_BACKGROUND;
    }};
  }

  &::-webkit-scrollbar {
    width: 8px;
    background-color: ${(props) => {
      return props.theme === 'day' ? 'rgba(0, 0, 0, 0.25)' : 'rgba(255, 255, 255, 0.25)';
    }};
  }

  &::-webkit-scrollbar-thumb {
    background-color: ${(props) => {
      return props.theme === 'day' ? 'rgba(0, 0, 0, 0.25)' : 'rgba(255, 255, 255, 0.25)';
    }};
    border-radius: 10px;
  }
`;

const InnerContent = (props) => {
  const { children, style, theme, } = props;
  return (
    <InnerContentStyle
      style={style}
      theme={theme}
    >
      {children}
    </InnerContentStyle>
  );
};

InnerContent.propTypes = {
  theme: PropTypes.string,
  style: PropTypes.object,
};

InnerContent.defaultProps = {
  theme: 'day',
  style: null,
};

export default InnerContent;
