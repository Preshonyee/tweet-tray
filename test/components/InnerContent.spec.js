import React from 'react';
import Enzyme, { shallow, } from 'enzyme';
import { shallowToJson, } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import InnerContent from '../../app/components/InnerContent';

Enzyme.configure({ adapter: new Adapter(), });

function setup() {
  const component = shallow(<InnerContent><div>Test Content</div></InnerContent>);
  return {
    component,
    innerContent: component.find('InnerContent__InnerContentStyle').find('div'),
  };
}

describe('InnerContent component', () => {
  it('should render', () => {
    const { component, } = setup();
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should render content', () => {
    const { innerContent, } = setup();
    expect(shallowToJson(innerContent)).toMatchSnapshot();
  });
});
