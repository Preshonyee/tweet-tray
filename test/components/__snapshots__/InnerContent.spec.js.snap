// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InnerContent component should render 1`] = `
<InnerContent__InnerContentStyle
  style={null}
  theme="day"
>
  <div>
    Test Content
  </div>
</InnerContent__InnerContentStyle>
`;

exports[`InnerContent component should render content 1`] = `
<div>
  Test Content
</div>
`;
